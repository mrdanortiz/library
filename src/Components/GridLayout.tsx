import React, { FunctionComponent } from "react";
import { Container, Grid, Image, SemanticWIDTHS } from "semantic-ui-react";

interface Props {
    splashImages: Array<{
        id: string;
        alt_description: string;
        urls: {
            raw: string;
            full: string;
            regular: string;
            small: string;
        };
    }>;
}

const GridLayout: FunctionComponent<Props> = ({ splashImages }) => (
    <Container>
        {splashImages ? (
            <Grid>
                <Grid.Row columns={((splashImages && splashImages.length) as SemanticWIDTHS) || 1}>
                    {splashImages &&
                        splashImages.map(img => (
                            <Grid.Column key={img.id}>
                                <Image src={img.urls.small} alt={img.alt_description} />
                            </Grid.Column>
                        ))}
                </Grid.Row>
            </Grid>
        ) : (
            undefined
        )}
    </Container>
);

export default GridLayout;
