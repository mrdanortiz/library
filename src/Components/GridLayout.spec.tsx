import React from "react";
import renderer from "react-test-renderer";
import { GridLayout } from "./GridLayout";

const splashImages = [
    {
        "id": "dO2WTawCTC4",
        "alt_description": "A gray and white husky drinking water out of a Boxed Water carton",
        "urls": {
            "raw": "https://images.unsplash.com/photo-1553531384-411a247ccd73?ixlib=rb-1.2.1&ixid=eyJhcHBfaWQiOjYzNTYzfQ",
            "full": "https://images.unsplash.com/photo-1553531384-411a247ccd73?ixlib=rb-1.2.1&q=85&fm=jpg&crop=entropy&cs=srgb&ixid=eyJhcHBfaWQiOjYzNTYzfQ",
            "regular": "https://images.unsplash.com/photo-1553531384-411a247ccd73?ixlib=rb-1.2.1&q=80&fm=jpg&crop=entropy&cs=tinysrgb&w=1080&fit=max&ixid=eyJhcHBfaWQiOjYzNTYzfQ",
            "small": "https://images.unsplash.com/photo-1553531384-411a247ccd73?ixlib=rb-1.2.1&q=80&fm=jpg&crop=entropy&cs=tinysrgb&w=400&fit=max&ixid=eyJhcHBfaWQiOjYzNTYzfQ",
            "thumb": "https://images.unsplash.com/photo-1553531384-411a247ccd73?ixlib=rb-1.2.1&q=80&fm=jpg&crop=entropy&cs=tinysrgb&w=200&fit=max&ixid=eyJhcHBfaWQiOjYzNTYzfQ"
        }
    }
];

it("renders correctly", () => {
    const tree = renderer.create(<GridLayout splashImages={splashImages} />).toJSON();
    expect(tree).toMatchSnapshot();
});
