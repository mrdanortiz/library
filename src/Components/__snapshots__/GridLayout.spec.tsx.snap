// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly 1`] = `
<div
  className="ui container"
>
  <div>
    <div
      className="ui center aligned middle aligned grid"
      style={
        Object {
          "height": "100%",
        }
      }
    >
      <div
        className="one column row"
      >
        <div
          className="eight wide computer sixteen wide mobile column"
        >
          <img
            alt="A gray and white husky drinking water out of a Boxed Water carton"
            className="ui medium bordered circular centered image"
            src="https://images.unsplash.com/photo-1553531384-411a247ccd73?ixlib=rb-1.2.1&ixid=eyJhcHBfaWQiOjYzNTYzfQq=40&fit=crop&w=300&h=300"
          />
        </div>
      </div>
    </div>
  </div>
</div>
`;
